### PR TITLE
assert grpc call errors

### PR DESCRIPTION
**Pull request by eguzki**
Expect from the GRPC call to return error as well.

### CODE REVIEW

File: src/expect_interface.rs [hunk 1]
```diff
@@ -182,15 +183,15 @@ impl<'a> ExpectGrpcCall<'a> {
         }
     }
 
-    pub fn returning(&mut self, token_id: Option<u32>) -> &mut Tester {
+    pub fn returning(&mut self, res: Result<u32, Status>) -> &mut Tester {
```
Comment (alexsnaps) [inline src/expect_interface.rs:186]: Don't think this should be a `Result` or then a `Option<Result>`... didn't look to closely into this particular expect, but the pattern was (tho arguably confusing), that it'd only validate when a `Some` was provided, while `None` would mean: while I expect this call, I don't care what it gets called with...

File: src/hostcalls.rs [hunk 2]
```diff
@@ -1506,25 +1506,34 @@ fn get_hostfunc(
...
+                        .staged
+                        .get_expect_grpc_call(
+                            service,
+                            service_name,
+                            method_name,
+                            initial_metadata,
+                            request,
+                            timeout_milliseconds,
+                        )
+                        .unwrap_or(Ok(0));
```
Comment (alexsnaps) [inline src/hostcalls.rs:1521]: You seem to change this for a `Err` now, to get the previous behavior... that was already confusing, but that's now a new pattern, I don't think this makes sense...

File: src/expect_interface.rs (hunk 1 above)
Comment (alexsnaps) [reply src/expect_interface.rs:186]: Nevermind that's the return

File: src/hostcalls.rs (hunk 2 above)
Comment (alexsnaps) [reply src/hostcalls.rs:1521]: So why the `Result` here? if the `Err` path gets dropped on the floor?

File: src/hostcalls.rs [hunk 3]
```diff
@@ -1506,25 +1506,34 @@ fn get_hostfunc(
...
+
+                    if let Ok(token_id) = call_result {
```
Comment (alexsnaps) [inline src/hostcalls.rs:1523]: This path will always hit, no?

File: src/hostcalls.rs [hunk 4]
```diff
@@ -1506,25 +1506,34 @@ fn get_hostfunc(
...
+                        unsafe {
+                            let return_token_add = mem.data_mut(&mut caller).get_unchecked_mut(
+                                token_ptr as u32 as usize..token_ptr as u32 as usize + 4,
+                            );
+                            return_token_add.copy_from_slice(&token_id.to_le_bytes());
+                        }
                     }
 
+                    let call_status = match call_result {
```
Comment (alexsnaps) [inline src/hostcalls.rs:1532]: also, what's with this? Isn't this _always_ `Ok(some int)`?

File: src/hostcalls.rs (hunk 2 above)
Comment (alexsnaps) [reply src/hostcalls.rs:1521]: It get rewrapped in an `Option::Some` always... Now I get it

Comment (alexsnaps) [reply src/hostcalls.rs:1521]: ... other than when it isn't expected...

**Review by alexsnaps — APPROVED** [review]

**Comment by eguzki** [comment]
I was thinking we should bring these changes to https://github.com/proxy-wasm/test-framework/pull/17

wdyt @alexsnaps ?

**Pull request merged**